source/slang/slang-ir-hlsl-legalize.cpp

```
@@ -6,12 +6,151 @@
-#include <functional>
+#include "slang-rich-diagnostics.h"
+#include "slang-type-system-shared.h"
 
 namespace Slang
 {
 
+bool isBarrierFlagGetterOp(IROp op)
+{
+    switch (op)
+    {
+    case kIROp_GetEnumBarrierMemoryTypeFlags:
```

Cppcheck (2.21.0)

[error] There is an unknown macro here somewhere. Configuration is required. If FIDDLE is a macro then please configure it.

(unknownMacro)